extension/core_functions/scalar/date/functions.json

```diff
@@ -188,9 +188,18 @@
     },
     {
         "name": "make_date",
-        "parameters": "year,month,day",
-        "description": "The date for the given parts",
-        "example": "make_date(1992, 9, 20)",
+        "variants": [
+            {
+                "parameters": "year,month,day",
+                "description": "The date for the given parts",
+                "example": "make_date(1992, 9, 20)"
+            },
+            {
+                "parameters": "date-struct::STRUCT(year BIGINT, month BIGINT, day BIGINT)",
```

Comment:
So, since JSON is a structured format, doesn't it make more sense to store the separate parameters and types as an array/struct instead? I think it would be much easier to parse and author (even if we end up smashing everything together in the generated headers later anyway) e.g.

```json
{
                "parameter_names": ["year", "month","day"],
                "parameter_types": ["INT", "INT", "INT"]
                "description": "The date for the given parts",
                "example": "make_date(1992, 9, 20)"
},
```

Or maybe even 

```json
{
                "parameters": [
                          {"name": "year", "type": "INT"},  
                          {"name": "month", "type": "INT"},
                          {"name": "day", "type": "INT"}
                 ],
                "description": "The date for the given parts",
                "example": "make_date(1992, 9, 20)"
},
```

This could allow us to also document stuff like "default" parameters or additional info on a per-parameter basis in the future (maybe even per-parameter descriptions) like:

```json
....
 {"name": "month", "type": "INT", "info": "Must be between 1 and 12, cannot be NULL!" },
 ....
```

Reply:
Yeah its also how the c-api is defined

Reply:
done: name is mandatory, type is optional for now. If type is missing <ANY> is implied.